test/powershell/Modules/Microsoft.PowerShell.Management/TimeZone.Tests.ps1

```diff
@@ -35,71 +30,83 @@ Describe "Get-Timezone test case no switches" -Tags "CI" {
     }
 }
 
 Describe "Get-Timezone test cases" -Tags "CI" {
 
+    BeforeAll {
+        $TimeZonesAvailable = [System.TimeZoneInfo]::GetSystemTimeZones()
+    }
+
     It "Call without ListAvailable switch returns an object of type TimeZoneInfo" {
         $result = Get-TimeZone
         $result | Should BeOfType TimeZoneInfo
     }
 
     It "Call WITH ListAvailable switch returns ArrayList of TimeZoneInfo objects where the list is greater than 0 item" {
         $list = Get-TimeZone -ListAvailable
-        $list.Count -gt 0 | Should Be $true
+        $list.Count | Should BeGreaterThan 0
 
         ,$list | Should BeOfType "Object[]"
         $list[0] | Should BeOfType "TimeZoneInfo"
     }
 
+    ## The local time zone could be set to UTC or GMT*. In this case, the .NET API returns the region ID
+    ## and not 'UTC'. To avoid a string matching error, we compare the BaseUtcOffset instead.
     It "Call with ListAvailable switch returns a list containing TimeZoneInfo.Local" {
-        $observedIdList = Get-TimeZone -ListAvailable | Select-Object -ExpandProperty Id
-        $oneExpectedId = ([System.TimeZoneInfo]::Local).Id
-        $observedIdList -contains $oneExpectedId | Should Be $true
+        $observedIdList = Get-TimeZone -ListAvailable | Select-Object -ExpandProperty BaseUtcOffset
+        $oneExpectedOffset = ([System.TimeZoneInfo]::Local).BaseUtcOffset
+        $observedIdList -eq $oneExpectedOffset | Should Be $oneExpectedOffset
     }
 
+    ## The local time zone could be set to UTC or GMT*. In this case, the .NET API returns the region ID
+    ## and not UTC. To avoid a string matching error, we compare the BaseUtcOffset instead.
     It "Call with ListAvailable switch returns a list containing one returned by Get-TimeZone" {
-        $observedIdList = Get-TimeZone -ListAvailable | Select-Object -ExpandProperty Id
-        $oneExpectedId = (Get-TimeZone).Id
-        $observedIdList -contains $oneExpectedId | Should Be $true
+        $observedIdList = Get-TimeZone -ListAvailable | Select-Object -ExpandProperty BaseUtcOffset
+        $oneExpectedOffset = (Get-TimeZone).BaseUtcOffset
+        $observedIdList -eq $oneExpectedOffset | Should Be $oneExpectedOffset
     }
 
     It "Call Get-TimeZone using ID param and single item" {
-        (Get-TimeZone -Id "Cape Verde Standard Time").Id -eq "Cape Verde Standard Time" | Should Be $true
+        $selectedTZ = $TimeZonesAvailable | Select-Object -Index ((Get-Random) % $TimeZonesAvailable.Count)
+        (Get-TimeZone -Id $selectedTZ.Id).Id | Should Be $selectedTZ.Id
     }
 
     It "Call Get-TimeZone using ID param and multiple items" {
-        $idList = @("Cape Verde Standard Time","Morocco Standard Time","Azores Standard Time")
-        $result = (Get-TimeZone -Id $idList).Id
-        Assert-ListsSame $result $idList
+        $randomIndexes = ((Get-Random) % $TimeZonesAvailable.Count), ((Get-Random) % $TimeZonesAvailable.Count),((Get-Random) % $TimeZonesAvailable.Count)
```

Comment:
>Get-Random [](start = 27, length = 10)

also, if you're going to use get-random, it might be more efficient to do:
`get-random -input (1..$TimeZonesAvailable.Count) -count 3`